ripcord: init at 0.4.18

MDeltaX opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

> Ripcord is a desktop chat client for group-centric services like Slack and Discord. It provides a traditional compact desktop interface designed for power users. It's not built on top of web browser technology: it responds quickly to input, sips gently from computer resources, and gets out of your way. It does voice chat, too.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @ldesgoui @MP2E @tadeokondrak

MDeltaX committed:
ripcord: init at 0.4.18

tadeokondrak commented on pkgs/applications/networking/instant-messengers/ripcord/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  meta = with stdenv.lib; {
+    description = "A desktop chat client for Discord and Slack";
+    homepage = "https://cancel.fm/ripcord/";
```
Is it really redistributable?

MicrosoftTakeover replied:
The author has stated in Discord that it's "all rights reserved but redistribution without modifications is allowed (shareware)"
https://discordapp.com/channels/231664689694638081/231664689694638081/467792133319950337

MDeltaX replied:
yep

MicrosoftTakeover replied:
https://cancel.fm/ripcord/shareware-redistribution/

tadeokondrak commented on pkgs/applications/networking/instant-messengers/ripcord/default.nix:
```diff
@@ -0,0 +1,32 @@
+in appimageTools.wrapType2 rec {
+  inherit name src;
+
+  extraInstallCommands = ''
```
I think we should wrap the binary to set `RIPCORD_ALLOW_UPDATES=0` so it doesn't try to automatically update itself and fail

HeapUnderfl0w replied:
afaik ripcord never attempts to self update. Ripcord only queries the update URL to show update notifications in Ripcord itself (which the `RIPCORD_ALLOW_UPDATES` disables).

MicrosoftTakeover replied:
I personally would not like update notifications for software delivered via package manger. You end up in annoying situations where there's a new version released but it's not packaged yet so you get notified over and over.

alexarice commented on maintainers/maintainer-list.nix:
```diff
@@ -4101,6 +4101,12 @@
     githubId = 1377571;
     name = "Matthew S. Daiter";
   };
+  MDeltaX = {
```
The addition to `maintainer-list.nix` should be in a separate commit named `maintainers: add MDeltaX`

alexarice commented on pkgs/applications/networking/instant-messengers/ripcord/default.nix:
```diff
@@ -0,0 +1,32 @@
+    inherit name src;
+  };
+in appimageTools.wrapType2 rec {
+  inherit name src;
```
Does this not accept `pname`?

MDeltaX replied:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/appimage/default.nix#L56
apparently nope

alexarice commented on pkgs/applications/networking/instant-messengers/ripcord/default.nix:
```diff
@@ -0,0 +1,32 @@
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://cancel.fm/dl/${pname}-${version}-x86_64.AppImage";
```
```suggestion
    url = "https://cancel.fm/dl/${name}-x86_64.AppImage";
```
Since you've defined name anyway

alexarice commented on pkgs/applications/networking/instant-messengers/ripcord/default.nix:
```diff
@@ -0,0 +1,32 @@
+    install -m 444 -D ${appimageContents}/Ripcord_Icon.png $out/share/icons/hicolor/512x512/apps/Ripcord_Icon.png
+  '';
+
+  meta = with stdenv.lib; {
```
As far as I can see you do not use `stdenv` anywhere else so you could just import `lib` at the top instead

```suggestion
  meta = with lib; {
```

Commit:
Update pkgs/applications/networking/instant-messengers/ripcord/default.nix

oh didn't know you could use buttons for this. neat.

yeah, just copy pasted some existing appImage expression. can you recommend a linter/beautifier for nix?

Co-Authored-By: Alex Rice <alexrice999@hotmail.co.uk>

Commit:
Update pkgs/applications/networking/instant-messengers/ripcord/default.nix

Co-Authored-By: Alex Rice <alexrice999@hotmail.co.uk>

MDeltaX commented:
will come back to this in the next days

Profpatsch commented:
Is this ready for merge?

colemickens commented:
Hey @MDeltaX, do you think you'll get to this soon, and if not, do you mind if someone else pushes it along? Thanks for doing this work!

colemickens commented:
Does this need some Qt wrap hooks? I'm not able to run this on Wayland due to the usual Qt platform plugin err.

bqv commented:
Christ this is old. Sorry this nobody ever bothered to merge this. My version is more up to date at this point anyway

Profpatsch commented:
Then let’s close in favor of the other PR.
